x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -508,6 +522,89 @@ protected void onAbort() {
         context.shutdown();
     }
 
+    /**
+     * Let the indexer stop at the next checkpoint and call the listener after the flag has been persisted in state.
+     *
+     * If the indexer isn't running, persist state if required and call the listener immediately.
+     */
+    synchronized void setStopAtCheckpoint(boolean shouldStopAtCheckpoint, ActionListener<Void> shouldStopAtCheckpointListener) {
+        IndexerState state = getState();
+
+        // in case the indexer isn't running, respond immediately
+        if (state == IndexerState.STARTED && context.shouldStopAtCheckpoint() != shouldStopAtCheckpoint) {
+            context.setShouldStopAtCheckpoint(shouldStopAtCheckpoint);
+
+            // because save state is async we need to block the call until state is persisted, so that the job can not
+            // be triggered
+            CountDownLatch latch = new CountDownLatch(1);
```

Comment:
Could you adjust the comment indicating that this is because the method is `synchronized`? It took me a second to see that all the triggering methods, etc. were synchronized. 

Also, it might be good to make this method `final` so it cannot be overridden and consequently cause MT issues in the future by accident.